models/google_ads__utm_report.sql

```diff
@@ -0,0 +1,89 @@
+with stats as (
+
+    select *
+    from {{ var('ad_stats') }}
+
+), accounts as (
+
+    select *
+    from {{ var('account_history') }}
+    where is_most_recent_record = True
+    
+), campaigns as (
+
+    select *
+    from {{ var('campaign_history') }}
+    where is_most_recent_record = True
+    
+), ad_groups as (
+
+    select *
+    from {{ var('ad_group_history') }}
+    where is_most_recent_record = True
+    
+), ads as (
+
+    select *
+    from {{ var('ad_history') }}
+    where is_most_recent_record = True
+    
+), fields as (
+
+    select
+        stats.date_day,
+        accounts.account_name,
+        accounts.account_id,
+        campaigns.campaign_name,
+        campaigns.campaign_id,
+        ad_groups.ad_group_name,
+        ad_groups.ad_group_id,
+        ads.ad_id,
+        ads.base_url,
+        ads.url_host,
+        ads.url_path,
+
+        {% if var('google_auto_tagging_enabled', false) %}
+
+        coalesce( {{ dbt_utils.get_url_parameter('ads.final_url', 'utm_source') }} , 'google')  as utm_source,
+        coalesce( {{ dbt_utils.get_url_parameter('ads.final_url', 'utm_medium') }} , 'cpc') as utm_medium,
+        coalesce( {{ dbt_utils.get_url_parameter('ads.final_url', 'utm_campaign') }} , campaigns.campaign_name) as utm_campaign,
+        coalesce( {{ dbt_utils.get_url_parameter('ads.final_url', 'utm_content') }} , ad_groups.ad_group_name) as utm_content,
+
+        {% else %}
+
+        ads.utm_source,
+        ads.utm_medium,
+        ads.utm_campaign,
+        ads.utm_content,
+        
+        {% endif %}
+
+        ads.utm_term,
+        sum(stats.spend) as spend,
+        sum(stats.clicks) as clicks,
+        sum(stats.impressions) as impressions
+
+        {% for metric in var('google_ads__ad_stats_passthrough_metrics', []) %}
+        , sum(stats.{{ metric }}) as {{ metric }}
+        {% endfor %}
+
+    from stats
+    left join ads
+        on stats.ad_id = ads.ad_id
+        and stats.ad_group_id = ads.ad_group_id
+    left join ad_groups
+        on ads.ad_group_id = ad_groups.ad_group_id
+    left join campaigns
+        on ad_groups.campaign_id = campaigns.campaign_id
+    left join accounts
+        on campaigns.account_id = accounts.account_id
+    
+    -- I WANT OTHER EYES ON THIS TO VALIDATE!
+    -- We only want utm ads to populate this report. Therefore, we filter where url ads are populated.
+    where ads.source_final_urls is not null
```

Comment:
I have a few questions here:
1. Would `ads.final_url` be different than `ads.source_final_urls`?

2. I noticed in FB that there could be URLs however the UTM fields could still be null. In which sense, I'd like to highlight my question above here as well
> Is there a fundamental difference between UTMs and URLs?

So if that is the case, should we be doing the non-null condition on UTMs extracted fields or URLs?

3. To make sure it filters for both the case of auto tagging enabled and not enabled, would it be more straight forward to do a `filtered_fields` CTE after this `fields` CTE for the below conditions instead?
`
    where utm_source is not null 
        or utm_medium is not null
        or utm_campaign is not null 
        or utm_content is not null
`

4. Additionally, is there a concept of `utm_term` for Google Ads? FB has 5 total.

Reply:
> Would ads.final_url be different than ads.source_final_urls?

- `final_url` is the modified version of `final_urls`, whereas `source_final_urls` is the original version of `final_urls`. I felt that including the filter on the original field vs. the field we applied the concatenation would be better.

> Questions 2 and 3

- I think we should think more about the fundamental difference between UTMs and URLs. If there is a fundamental difference then I think we should go the route of filtering for only UTM populated fields. However, if there isn't then I think we should keep this filter as is and rename the model to be `google_ads__url_report`. I am becoming more keen to this idea. Especially as our original model for the ad reporting model was built for `url ad adapter` models. 🤔 Let's chat more about this one.

> Additionally, is there a concept of utm_term for Google Ads? FB has 5 total.
 
- Yes, Google has this concept. However, the auto tagging does't use this.